fence/README.md

```diff
@@ -0,0 +1,153 @@
+  rpc EnableClusterFence (EnableClusterFenceRequest)
+  returns (EnableClusterFenceResponse) {}
+
+  // DisableClusterFence RPC call to remove a list of CIDR blocks from the
+  // list of blocklisted/fenced clients.
+  rpc DisableClusterFence (DisableClusterFenceRequest)
+  returns (DisableClusterFenceResponse) {}
+
+  // ListClusterFence RPC call to provide a list of blocklisted/fenced clients.
+  rpc ListClusterFence(ListClusterFenceRequest)
+  returns (ListClusterFenceResponse){}
+}
+```
+
+### EnableClusterFence
+
+```protobuf
+// EnableClusterFenceRequest contains the information needed to identify
+// the cluster so that the appropriate fencing operation can be
+// performed.
+message EnableClusterFenceRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(fence_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDRList cidr_list = 3;
+}
+
+// EnableClusterFenceResponse holds the information about the result of the
+// EnableClusterFenceRequest call.
+message EnableClusterFenceResponse {
+    // The message returned on execution as received from SP.
+    string result = 1;
+}
+```
+
+### DisableClusterFence
+
+```protobuf
+// DisableClusterFenceRequest contains the information needed to identify
+// the cluster so that the appropriate fence operation can be
+// disabled.
+message DisableClusterFenceRequest {
+  // Plugin specific parameters passed in as opaque key-value pairs.
+  map<string, string> parameters = 2;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 3 [(fence_secret) = true];
+  // list of CIDR blocks on which the fencing operation is expected to be
+  // performed.
+  repeated CIDRList cidr_list = 4;
+}
+
+// DisableClusterFenceResponse holds the information about the result of the
+// DisableClusterFenceRequest call.
+message DisableClusterFenceResponse {
+    // The message returned on execution as received from SP.
+    string result = 1;
+}
+```
+### ListClusterFence
+
+```protobuf
+// ListClusterFenceRequest contains the information needed to identify
+// the cluster so that the appropriate fenced clients can be listed.
+message ListClusterFenceRequest {
+    // Plugin specific parameters passed in as opaque key-value pairs.
+    map<string, string> parameters = 1;
+    // Secrets required by the plugin to complete the request.
+    map<string, string> secrets = 2 [(fence_secret) = true];
+}
+
+// ListClusterFenceResponse holds the information about the result of the
+// ListClusterFenceResponse call.
+message ListClusterFenceResponse {
+    // list of IPs that are blocklisted by the SP.
+    repeated CIDRList cidr_list = 1;
+}
+```
+
+### CIDR blocks
+
+```protobuf
+// CIDRList is a list of CIDR blocks.
+message CIDRList {
+  string cidr = 1;
```

Comment:
This is not a list, it is a single `cidr`. Does it need to be `repeated`, and drop the `repeated` keyword from `ListClusterFenceResponse`?

Reply:
Sure, I agree. Although the CIDRList message is supposed to be a list of strings, the naming convention is a bit off.
Changing the name from CIDRList to CIDR might better describe what we are trying to do.
Does this look better?
```
// ListClusterFenceResponse holds the information about the result of the
// ListClusterFenceRequest call.
message ListClusterFenceResponse {
  // list of CIDR blocks that are blocklisted by the SP.
  repeated CIDR cidrs = 1;
}

message CIDR {
  // CIDR blocks
  string cidr = 1;
}
```

Reply:
yes, this approach looks better to me.